games/devtest/mods/unittests/misc.lua

```diff
@@ -68,3 +68,25 @@ local function test_clear_meta(_, pos)
 	end
 end
 unittests.register("test_clear_meta", test_clear_meta, {map=true})
+
+local function test_compress()
+	-- This text should be compressible, to make sure the results are... normal
+	local text = "The\000 icey canoe couldn't move very well on the\128 lake. The\000 ice was too stiff and the icey canoe's paddles simply wouldn't punch through."
+	local methods = {"deflate"}
+	if not core.features.compress_zstd then
+		core.log("warning", "test_compress: Server too old, skipping zstd test.")
+	else
+		table.insert(methods, "zstd")
+		-- table.insert(methods, "noodle") -- for warning alarm test
+	end
+	for _, method in ipairs(methods) do
+		local compressed = core.compress(text, method)
+		assert(core.decompress(compressed, method) == text, "input/output mismatch for compression method " .. method)
+		if method == "zstd" then
+			assert(compressed:sub(1, 4) == "\040\181\047\253", "zstd magic number not in zstd output")
```

Comment:
If you wanted to use hex here, you could use `string.char(0x..., 0x..., 0x..., 0x...)`. You could shorten the assertions to `assert((method == "zstd") == (compressed:sub(1, 4) == "\040\181\047\253"))`.

Reply:
I've partially introduced these changes in 1032f7b but I'm keeping differing error messages depending on the exact failure (i.e. specific method) unless explicitly told otherwise.